src/graph/util/ExpressionUtils.cpp

```diff
@@ -536,7 +536,61 @@ StatusOr<Expression *> ExpressionUtils::foldConstantExpr(const Expression *expr)
     }
     return foldedExpr;
   }
-  return newExpr;
+
+  auto matcher = [](const Expression *e) {
+    return e->kind() == Expression::Kind::kLogicalAnd || e->kind() == Expression::Kind::kLogicalOr;
+  };
+  auto rewriter = [](const Expression *e) {
+    auto logicalExpr = static_cast<const LogicalExpression *>(e);
+    return simplifyLogicalExpr(logicalExpr);
+  };
+  return RewriteVisitor::transform(newExpr, matcher, rewriter);
+}
+
+Expression *ExpressionUtils::simplifyLogicalExpr(const LogicalExpression *logicalExpr) {
+  auto *expr = static_cast<LogicalExpression *>(logicalExpr->clone());
+  if (expr->kind() == Expression::Kind::kLogicalXor) return expr;
+
+  ObjectPool *objPool = logicalExpr->getObjPool();
+
+  // Simplify logical and/or
+  for (auto iter = expr->operands().begin(); iter != expr->operands().end(); ++iter) {
+    auto *operand = *iter;
+    if (operand->kind() != Expression::Kind::kConstant) continue;
+    auto &val = static_cast<ConstantExpression *>(operand)->value();
+    if (!val.isBool()) continue;
+    if (expr->kind() == Expression::Kind::kLogicalAnd) {
+      if (val.getBool()) {
+        // Remove the true operand
+        expr->operands().erase(iter);
```

Comment:
`iter = expr->operands().erase(iter); continue;` and remove the `--iter;`

Reply:
Ok